.github/workflows/shellcheck.yml

```diff
@@ -0,0 +1,16 @@
+on:
+  push:
+    branches:
+      - "*"
+
+name: "Shellcheck"
+
+jobs:
+  shellcheck:
+    name: Shellcheck
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Run ShellCheck
+        uses: ludeeus/action-shellcheck@master
```

Comment:
Pin the exact version of this third party `ludeeus/action-shellcheck` action with SHA. It would look like this for the current head:

`uses: ludeeus/action-shellcheck@b2bbefc2e6b9dcbc6355b85e366c9e55bf8d57e1`
 
See more at: https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions

Reply:
Oh, I hadn't realized that a push triggerd workflow will have a write-access capable GITHUB_TOKEN and a compromised lint action could write to the repo. Good catch.